src/main/java/com/kobylynskyi/graphql/codegen/mapper/ValueMapper.java

```diff
@@ -65,7 +67,28 @@ private static String mapBoolean(BooleanValue value) {
         return value.isValue() ? "true" : "false";
     }
 
-    private static String mapInt(IntValue value) {
+    private static String mapInt(MappingContext mappingContext, IntValue value, Type<?> graphQLType) {
+        //default java basic type is `int`. so, default value like 123 that must wrap or append suffix `L` when it be defined as `int` in graphql schema.
+        //`int` cannot assign to `Long`, also `double` cannot assign to `Float`, but graphql Float default mapping is Double in java, so, not modify `mapFloat`.
+        if (graphQLType != null) {
+            List<Field> fields = Arrays.stream(graphQLType.getClass().getDeclaredFields()).collect(Collectors.toList());
+            String customType = mappingContext.getCustomTypesMapping().get("Long");
+            for (Field f : fields) {
```

Comment:
@jxnu-liguobin 
1. Could you please explain what are you trying to get from `graphQLType fields`?
2. In `customTypesMapping` you are checking for a key `Long`, but someone can have scalar called `myLong` and a mapping `myLong -> java.lang.Long`. So need to check for a value and not the key.

Here's how I would write the whole method:
```java
    private static String mapInt(MappingContext mappingContext, IntValue value, Type<?> graphQLType) {
        if (graphQLType instanceof TypeName) {
            for (Map.Entry<String, String> entry : mappingContext.getCustomTypesMapping().entrySet()) {
                if (isLong(entry.getValue())) {
                    String typeName = ((TypeName) graphQLType).getName();
                    if (isLong(typeName)) {
                        return String.valueOf(value.getValue()).concat("L");
                    }
                }
            }
        }
        if (graphQLType instanceof NonNullType) {
            // unwrapping NonNullType
            return mapInt(mappingContext, value, ((NonNullType) graphQLType).getType());
        }
        return String.valueOf(value.getValue());
    }
```

Reply:
> @jxnu-liguobin
> 
> 1. Could you please explain what are you trying to get from `graphQLType fields`?
> 2. In `customTypesMapping` you are checking for a key `Long`, but someone can have scalar called `myLong` and a mapping `myLong -> java.lang.Long`. So need to check for a value and not the key.
> 
> Here's how I would write the whole method:
> 
> ```java
>     private static String mapInt(MappingContext mappingContext, IntValue value, Type<?> graphQLType) {
>         if (graphQLType instanceof TypeName) {
>             for (Map.Entry<String, String> entry : mappingContext.getCustomTypesMapping().entrySet()) {
>                 if (isLong(entry.getValue())) {
>                     String typeName = ((TypeName) graphQLType).getName();
>                     if (isLong(typeName)) {
>                         return String.valueOf(value.getValue()).concat("L");
>                     }
>                 }
>             }
>         }
>         if (graphQLType instanceof NonNullType) {
>             // unwrapping NonNullType
>             return mapInt(mappingContext, value, ((NonNullType) graphQLType).getType());
>         }
>         return String.valueOf(value.getValue());
>     }
> ```

1. oh, it can use forced conversion instead of reflect, thanks.
2. beacuse i only want to solve `Long` in [graphql-java-scalar](https://github.com/graphql-java/graphql-java-extended-scalars), https://github.com/graphql-java/graphql-java-extended-scalars/blob/6f82be9d81b00a580f79efc12b47a0970b12f768/src/main/java/graphql/scalars/ExtendedScalars.java#L213 not customtype like myLong,xxLong.If you think this is better, i can change it.

Reply:
To 2: In case clients are using custom scalar Long, they will need to specify it in the customTypesMapping anyway. Otherwise all unknown scalars are mapped to String by default